src/semantic.py

```diff
@@ -94,12 +97,19 @@ def define_method(self, name: str, param_names: list, param_types: list, return_
             name, param_names, param_types, return_type)
         return method
 
+    def get_all_attributes(self):
+        all_attributes = []
```

Comment:
you could initialize this in this way: `all_attributes = self.parent and self.parent.get_all_attributes() or []` and remove the if stamtement